cpp/src/strings/contains.cu

```diff
@@ -0,0 +1,244 @@
+{
+    return detail::matches_re(strings, pattern, mr);
+}
+
+namespace detail
+{
+
+namespace
+{
+
+/**
+ * @brief This counts the number of times the regex pattern matches in each string.
+ *
+ */
+template<size_t stack_size>
+struct count_fn
+{
+    reprog_device prog;
+    column_device_view d_strings;
+
+    __device__ int32_t operator()(unsigned int idx)
+    {
+        u_char data1[stack_size], data2[stack_size];
+        prog.set_stack_mem(data1,data2);
+        if( d_strings.is_null(idx) )
+            return 0;
+        string_view d_str = d_strings.element<string_view>(idx);
+        int32_t find_count = 0;
+        size_type nchars = d_str.length();
+        size_type begin = 0;
+        while( begin <= nchars )
+        {
+            auto end = nchars;
+            if( prog.find(idx,d_str,begin,end) <=0 )
+                break;
+            ++find_count;
+            begin = end > begin ? end : begin + 1;
+        }
+        return find_count;
+    }
+};
+
+}
+
+std::unique_ptr<column> count_re( strings_column_view const& strings,
+                                  std::string const& pattern,
+                                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                  cudaStream_t stream = 0)
+{
+    auto strings_count = strings.size();
+    auto strings_column = column_device_view::create(strings.parent(),stream);
+    auto d_column = *strings_column;
+
+    // compile regex into device object
+    auto prog = reprog_device::create(pattern,get_character_flags_table(),strings_count,stream);
+    auto d_prog = *prog;
+
+    // create the output column
+    auto results = make_numeric_column( data_type{INT32}, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto d_results = results->mutable_view().data<int32_t>();
+
+    // fill the output column
+    auto execpol = rmm::exec_policy(stream);
+    int regex_insts = d_prog.insts_counts();
+    if( (regex_insts > MAX_STACK_INSTS) || (regex_insts <= RX_SMALL_INSTS) )
+        thrust::transform(execpol->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_count),
+            d_results, count_fn<RX_STACK_SMALL>{d_prog, d_column} );
+    else if( regex_insts <= RX_MEDIUM_INSTS )
+        thrust::transform(execpol->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_count),
+            d_results, count_fn<RX_STACK_MEDIUM>{d_prog, d_column} );
+    else
+        thrust::transform(execpol->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_count),
+            d_results, count_fn<RX_STACK_LARGE>{d_prog, d_column} );
+
+    results->set_null_count(strings.null_count());
```

Comment:
Why null_count be set twice ?

Reply:
Calling `mutable_view()` on a `column` invalidates the `null_count`. This sets it back correctly.